cmd/deploy.go

```diff
@@ -278,7 +285,133 @@ func deploy(
 	}
 
 	return nil
+}
+
+// This doesnt handle the option where a task is gracefully shutdown
+func didPipelineGetCancelled(
+	deployStatus *svcmgmtv1alpha1.DeployStatus,
+) bool {
+	if deployStatus == nil {
+		return false
+	}
+	if deployStatus.Succeeded != nil && deployStatus.Succeeded.Reason == "Cancelled" {
+		return true
+	}
+	for _, taskStatus := range deployStatus.DeployTaskStatus {
+		if taskStatus.CompletionTime == nil {
+			continue
+		}
+		for _, step := range taskStatus.Steps {
+			terminatedState := step.GetTerminated()
+			if terminatedState == nil {
+				continue
+			}
+			if terminatedState.Reason == "TaskRunCancelled" {
+				return true
+			}
+		}
+	}
+	return false
+}
 
+func streamPodErrorLogs(
+	ctx context.Context,
+	client svcmgmtv1alpha1.ServiceMgmtServiceClient,
+	envName string,
+	serviceName string,
+	deployStatus *svcmgmtv1alpha1.DeployStatus,
+) error {
+	if deployStatus == nil {
+		return fmt.Errorf("deploystatus was nil")
+	}
+
+	logrequest, err := getLogRequestFromDeployStatus(deployStatus)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("====================")
+	fmt.Printf("Printing logs for Task '%s' at Step '%s'\n\n", logrequest.TaskName, logrequest.TaskStep)
+	fmt.Println("====================")
+
+	stream, err := client.GetDeployLogs(ctx, &svcmgmtv1alpha1.GetDeployLogsRequest{
+		EnvironmentName: envName,
+		ServiceName:     serviceName,
+		LogRequest:      logrequest,
+	})
+	if err != nil {
+		return err
+	}
+	for {
+		response, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return err
+		}
+		fmt.Println(response.LogLine)
```

Comment:
surprised there's no stream.Close() or whatever here.. can't remember what hte api looks like, just curious.

Reply:
When the server gracefully closes the stream, the `err` object is equal to `io.EOF`

Reply:
The client only calls close in a bidirectional stream situation.

Reply:
![image](https://user-images.githubusercontent.com/2420177/212495602-994fd0f3-3461-400c-ade9-02534ac6706e.png)